RIOT-OS/RIOT
# gnrc_lorawan: cleanup PHY state machine

### Contribution description
This PR:
- Implements the PHY state machine of GNRC LoRaWAN using an FSM with entry/exit states. This simplifies the logic a lot, makes it more robust and facilitates adding other LoRaWAN classes.
- Cleans up timers and event functions, as those old function interfaces are not needed anymore.




### Testing procedure
Run `examples/gnrc_lorawan`. Make sure it's still possible to join a LoRaWAN network, send and receive data.



### Issues/PRs references
None so far, but it will be required for adding class C support (almost there!)
